python/tests/test_c_array.py

```diff
@@ -610,3 +622,84 @@ def test_c_array_duration():
         d2_duration_in_ms,
         d3_duration_in_ms,
     ]
+
+
+def test_device_array_errors(cuda_device):
+    if not cuda_device:
+        pytest.skip("CUDA device not available")
```

Comment:
Awesome! If you wanted to, I _think_ you could add this check in the pytest fixture. Either way, looks great to me!

Reply:
I've add this to tests in pyarrow already 😬 haha.

Reply:
Oops!